### DISCUSSION TITLE
I am unable to build without errors in Windows 11 using VS2022

### DISCUSSION THREAD

**Discussion by RallyTronics** [MethanePowered/MethaneAsteroids#5] [Q&A] [answered]
I am working in Windows 11 with VS2022.  I am using the build --vs2022 command and the build fails.  All of the standard MethaneKit examples work fine.  I have attached the logs.  Where did I go wrong?

[CMakeOutput.log](https://github.com/MethanePowered/MethaneAsteroids/files/10345354/CMakeOutput.log)
[CMakeError.log](https://github.com/MethanePowered/MethaneAsteroids/files/10345358/CMakeError.log)

**Comment by egorodet** [comment]
Hello @RallyTronics , sorry for late response (I did not notice your question in time). Please try to build the latest version again with script `Build\Windows\Build.bat --vs2022` - it works fine on my machine with the latest changes. If you still have a problem, please provide full build output with command `Build\Windows\Build.bat --vs2022 >Build.log 2>&1`, then attach `Build.log` to your message.

**Comment by RallyTronics** [comment]
[Build.log](https://github.com/MethanePowered/MethaneAsteroids/files/10716549/Build.log)
I gave `Build\Windows\Build.bat` --vs2022 a try and the build failed, I then ran `Build\Windows\Build.bat --vs2022 >Build.log 2>&1` and here is the build.log file.

**Comment by egorodet** [comment]
The attached `Build.log` is incomplete and is ending with `^CTerminate batch job (Y/N)? `, so it seems like the build process was terminated.

**Reply by RallyTronics** [reply]
[Build.log](https://github.com/MethanePowered/MethaneAsteroids/files/10716752/Build.log)

Sorry, this is the correct one

**Comment by RallyTronics** [comment]
[Build.log](https://github.com/MethanePowered/MethaneAsteroids/files/10716800/Build.log)
After cleaning then running `Build\Windows\Build.bat --vs2022 >Build.log 2>&1`

**Comment by egorodet** [comment, marked as answer]
As I see in Build.log, CMake configure fails with error “_download path exceeds the OS max path limit. The fully qualified file name must be less than 260 characters_”. So I suggest you trying to switch off [Windows path limit](https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=powershell), it can be done with the following command run from Administrator PowerShell:

```powershell
New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" `
-Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force
```

**Comment by RallyTronics** [comment]
Getting further looks like I am missing the `atlbase.h` header, I must be missing some prerequisite.  I will dig into it.

`D:\MethaneAsteroids\Build\Output\ExternalsCache\directxshadercompilerbinary\971179da0f7c0065f406274dc7442792d301d016\.\include\dxc/Support/WinIncludes.h(35,10): fatal  error C1083: Cannot open include file: 'atlbase.h': No such file or directory [D:\MethaneAsteroids\Build\Output\VisualStudio\x64-MSVC-DX-Release-SLN\Build\_deps\directxshadercompilerbinary-build\DirectXCompiler.vcxproj]
`

[Build.log](https://github.com/MethanePowered/MethaneAsteroids/files/10717611/Build.log)

**Reply by egorodet** [reply]
Yep, seems like you need to add "Visual C++ ATL Support" in your Visual Studio installation. See [Stack Overflow](https://stackoverflow.com/questions/3898287/c-include-atlbase-h-is-not-found) for more details.
